tests/shadow/test_transforms.py

```diff
@@ -0,0 +1,440 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the classical shadows transforms"""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+def hadamard_circuit(wires, shots=10000, interface="autograd"):
+    """Hadamard circuit to put all qubits in equal superposition (locally)"""
+    dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        for i in range(wires):
+            qml.Hadamard(wires=i)
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+def max_entangled_circuit(wires, shots=10000, interface="autograd"):
+    """maximally entangled state preparation circuit"""
+    dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        qml.Hadamard(wires=0)
+        for i in range(1, wires):
+            qml.CNOT(wires=[0, i])
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+def qft_circuit(wires, shots=10000, interface="autograd"):
+    """Quantum Fourier Transform circuit"""
+    dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+    one_state = np.zeros(wires)
+    one_state[-1] = 1
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        qml.BasisState(one_state, wires=range(wires))
+        qml.QFT(wires=range(wires))
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+def basic_entangler_circuit(wires, shots=10000, interface="autograd"):
```

Comment:
There is some ambiguity in the `wires` argument. Here it is an integer, and sometimes later it is a list, referring to the sub-space of qubits to create the state from. Can we a) call this n_wires and b) call the wires later something like `sub_wires` (or a better name)

Reply:
Can you link an example in the code where `wires` is the name of a list? I'm not entirely sure where the ambiguity is coming from.

Reply:
Lets say you want the states for `sub_wires = [[0,1], [1,2]]`, then you need the total wires to be `[0, 1, 2]` (I also would not `pytest.mark.parametrize` it in this case since it would unnecessarily do multiple quantum executions.

Reply:
Hmm ok. I'll change the name of this argument then. Later on we pass the "sub-wires" list directly to the `shadow.state` decorator so there isn't a new variable created there.